model/project_ref.go

```diff
@@ -654,19 +663,60 @@ func (p *ProjectRef) AttachToNewRepo(u *user.DBUser) error {
 			return errors.Wrap(err, "adding project to new repo scope")
 		}
 	}
+
 	update := bson.M{
-		"$set": bson.M{
-			ProjectRefOwnerKey:     p.Owner,
-			ProjectRefRepoKey:      p.Repo,
-			ProjectRefRepoRefIdKey: p.RepoRefId,
-		},
+		ProjectRefOwnerKey:     p.Owner,
+		ProjectRefRepoKey:      p.Repo,
+		ProjectRefRepoRefIdKey: p.RepoRefId,
 	}
-	if err := db.UpdateId(ProjectRefCollection, p.Id, update); err != nil {
+	update = p.addGithubConflictsToUpdate(update)
+	err = db.UpdateId(ProjectRefCollection, p.Id, bson.M{
+		"$set": update,
+	})
+	if err != nil {
 		return errors.Wrap(err, "updating owner/repo in the DB")
 	}
 	return GetAndLogProjectModified(p.Id, u.Id, false, before)
 }
 
+// addGithubConflictsToUpdate turns off any settings that may introduce conflicts by
+// adding fields to the given update and returning them.
+func (p *ProjectRef) addGithubConflictsToUpdate(update bson.M) bson.M {
+	// If the project ref doesn't default to repo, will just return the original project.
+	mergedProject, err := GetProjectRefMergedWithRepo(*p)
+	if err != nil {
+		grip.Debug(message.WrapError(err, message.Fields{
+			"message":            "unable to merge project with attached repo",
+			"project_id":         p.Id,
+			"project_identifier": p.Identifier,
+			"repo_id":            p.RepoRefId,
+		}))
+		return update
+	}
+	if mergedProject.IsEnabled() {
+		conflicts, err := mergedProject.GetGithubProjectConflicts()
+		if err != nil {
+			grip.Debug(message.WrapError(err, message.Fields{
+				"message":            "unable to get github project conflicts",
+				"project_id":         p.Id,
+				"project_identifier": p.Identifier,
+				"repo_id":            mergedProject.RepoRefId,
+			}))
+			return update
+		}
+		if len(conflicts.CommitQueueIdentifiers) > 0 {
+			update[bsonutil.GetDottedKeyName(projectRefCommitQueueKey, commitQueueEnabledKey)] = false
```

Comment:
maybe I'm not fully understanding what's happening here, but how do we know that the user doesn't want to explicitly have this set to true?

Reply:
Yes exactly, and that branch should take priority.

Alternatively, we could suppress moving repo for enabled projects overall, and that would remove the conflicts.

Reply:
Should we add some help text to the modal letting users know that these values will be automatically set?

Reply:
Sure I can just add that to the tooltip PR